src/core/models.py

```diff
@@ -104,7 +105,12 @@ def endpoints(self) -> str:
     @property
     def uris(self) -> str:
         """The ZooKeeper connection uris for the client application to connect with."""
-        return self._uris + self.chroot if self._uris else ""
+        # TODO (zkclient): Remove this property
+        warnings.warn(
```

Comment:
**suggestion** One could use the deprecated decorator, [here](https://pypi.org/project/Deprecated/).

Reply:
Do you prefer this specific implementation? Because we already have `typing-extensions` as a transitive dependency, which come with this [implementation](https://typing-extensions.readthedocs.io/en/latest/#typing_extensions.deprecated). We just need to properly add it as a dependency in `pyproject.toml

Reply:
Up to you guys. I generally used that decorator, and I like the decorator pattern (very functional). it should be a swift dependency. But up to you guys. It is not a strong point

Reply:
If that's ok, I'll use the `typing` one for the following reasons:
- already there, so no additional dep
- once we move on to python 3.13, it'll be part of the standard library so no change to the codebase except for the import statement

Reply:
Done in 7e48a95.
We needed to properly add `typing-extensions` to the project anyway, as we were already using `override` (part of the std lib in 3.12)

Reply:
But just to double check: does the deprecated decorator in the typing works both at runtime and with type checkers? 

A number of functions is the typing module only works with annotations/type checkers but don't enforce either validation or business logic at run time (which is what you were doing here and that also the deprecated library does)

Reply:
Yes, it works both at runtime and during [type checks](https://pyright-play.net/?strict=true&code=GYJw9gtgBALgngBwJYDsDmB9ApgDxllAZyTCKiQgTBBigBMsEQsBjAQ3zoCgeABBpqw5Y6ACgBE%2BQjHEBKLiwA2bQoSgAxMGABcXKPqi8mYBFhpw9B-o2btOEgCI2hMEiigQsqtmixzL%2BgzAUABGbCCihFiKwLJQALQAfOQoMLoGGVDMMACuIO4AzDyo0lAAvBpaovJhIOUp0gB0tTw8QA)

```text
# code in the link above

python -c "from foo import Foo;print(Foo().bar)"
<string>:1: DeprecationWarning: test
<string>:1: DeprecationWarning: Deprecation message
3
```

Reply:
Nice! TIL!!